### PR TITLE
Add check for pattern escapes in schema files

### PR DESCRIPTION
**Pull request by hermfischer-wf**
base spec conf test v401-05 check schema pattern element value for legal escapes

#### Reason for change
not passing v401-05 

#### Description of change
test pattern value escapes

#### Steps to Test
v401-05 

**review**:
@Arelle/arelle

### CODE REVIEW

**Comment by aviary2-wf** [comment]
## Security Insights
No security relevant content was detected by automated scans.

## Action Items
* Review PR for [security impact](https://wiki.atl.workiva.net/display/SECURITY/Development+Security+Review+Guidelines); comment "security review required" if needed or unsure
* Verify [`aviary.yaml`](https://wiki.atl.workiva.net/pages/viewpage.action?pageId=42762752#SecurityInsightFramework&Tooling-ConfiguringAviary) coverage of security relevant code
---
_Questions or Comments? Reach out on Slack: #support-infosec._

File: arelle/ValidateXbrlDTS.py [hunk 1]
```diff
@@ -46,6 +46,7 @@
 standard_roles_other = ("xbrl.5.1.3", ())
 
 inlineDisplayNonePattern = re.compile(r"display\s*:\s*none")
+illegalXsdPatternEscapeChar = re.compile(r"\\[^nrt\\.^?*+{}()[\]pPsSiIcCdDwW-]")
```
Comment (brettkail-wk) [inline arelle/ValidateXbrlDTS.py:49]: This matches `\|` but shouldn't (i.e., `<pattern value='\|'> matching literal `|` should be allowed).

This matches `\\z` but shouldn't (i.e., `<pattern value='\\z'>` matching the literal `\z` should be allowed).

Comment (hermfischer-wf) [reply arelle/ValidateXbrlDTS.py:49]: See https://www.w3.org/TR/2011/CR-xmlschema11-2-20110721/datatypes.diff-wd.html#cces
As I read the spec:
  * \\z  is not allowed
  * \\| is not allowed

Comment (brettkail-wk) [reply arelle/ValidateXbrlDTS.py]: @hermfischer-wf I agree `\z` is not allowed, but `\\z` should be (but this code incorrectly rejects it).

**Review by austinmatherne-wk — APPROVED** [review]

**Pull request merged**